clippy_utils/src/consts.rs

```diff
@@ -568,17 +632,30 @@ impl<'a, 'tcx> ConstEvalLateContext<'a, 'tcx> {
         match (l, r) {
             (Constant::Int(l), Some(Constant::Int(r))) => match *self.typeck_results.expr_ty_opt(left)?.kind() {
                 ty::Int(ity) => {
+                    let (ty_min_value, _) = ity.min_max()?;
+                    let bits = ity.bits();
                     let l = sext(self.lcx.tcx, l, ity);
                     let r = sext(self.lcx.tcx, r, ity);
+
+                    // Using / or %, where the left-hand argument is the smallest integer of a signed integer type and
+                    // the right-hand argument is -1 always panics, even with overflow-checks disabled
+                    if l == ty_min_value && r == -1 {
```

Comment:
I just realized that surely this is missing a condition like `matches!(op.node, BinOpKind::Div | BinOpKind::Rem)` (or should be moved inverted to the corresponding guards in the `match` below)?

Reply:
Huh yeah, this seems to unnecessarily bail out for the other ops where it would be ok. Not sure how I missed that when I wrote this, when the comment up top clearly mentions it 😅 
Will fix this in a bit